ddsrouter_core/src/cpp/writer/implementations/rtps/CommonWriter.cpp

```diff
@@ -237,6 +262,10 @@ void CommonWriter::internal_entities_creation_(
                       participant_id_ << " in topic " << topic_ << ".");
     }
 
+    // Set listener after entity creation to avoid SEGFAULT (produced when callback using rtps_writer_ is
+    // invoked before the variable is fully set)
+    rtps_writer_->set_listener(this);
+
```

Comment:
Please, set listener to null before destruction of the entity (for same reason).

NIT: Could you please check if this is done in Reader as well?